tests/compiler/test_arithemetic.py

```diff
@@ -0,0 +1,91 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import textwrap
+
+from inmanta import compiler
+
+
+def test_plus(snippetcompiler) -> None:
+    snippetcompiler.setup_for_snippet(
+        textwrap.dedent(
+            """
+            var1 = 1 + 2
```

Comment:
does it also work in expression? so the where clause of implementation, if, ...

Does is work in the body of a list comprehension?

Reply:
I'll extend the test case.

Reply:
Fix had to update the precedence rules to make it work in expressions. That's fixed now.